rockcraft.yaml

```diff
@@ -0,0 +1,183 @@
+  admin:
+    after: [base]
+    plugin: nil
+    build-environment:
+      - VERSION: "2.4.0"
+      - RANGER_TARGET: /home/ranger/dist/target
+      - RANGER_HOME: /usr/lib/ranger
+    override-build: |
+      craftctl default
+
+      # Create directories
+      mkdir -p \
+        ${CRAFT_PART_INSTALL}${RANGER_HOME}/admin \
+        ${CRAFT_PART_INSTALL}/var/run/ranger \
+        ${CRAFT_PART_INSTALL}/var/log/ranger \
+        ${CRAFT_PART_INSTALL}/usr/share/java
+
+      # Unpack admin tar
+      tar xvfz \
+        ${CRAFT_STAGE}${RANGER_TARGET}/ranger-${VERSION}-admin.tar.gz \
+        --directory=${CRAFT_PART_INSTALL}${RANGER_HOME}/admin \
+        --strip-components=1
+    stage:
+      - usr/lib/ranger/admin
+      - var/run/ranger
+      - var/log/ranger
+      - usr/share/java
+    permissions:
+      - path: usr/lib/ranger/admin
+        owner: 584792
+        group: 584792
+        mode: "755"
+      - path: var/run/ranger
+        owner: 584792
+        group: 584792
+        mode: "755"
+      - path: var/log/ranger
+        owner: 584792
+        group: 584792
+        mode: "755"
+      - path: usr/share/java
+        owner: 584792
+        group: 584792
+        mode: "755"
+
+  usersync:
+    after: [base]
+    plugin: nil
+    build-environment:
+      - VERSION: "2.4.0"
+      - RANGER_TARGET: /home/ranger/dist/target
+      - RANGER_HOME: /usr/lib/ranger
+    override-build: |
+      craftctl default
+
+      # Create directories
+      mkdir -p \
+        ${CRAFT_PART_INSTALL}${RANGER_HOME}/usersync \
+        ${CRAFT_PART_INSTALL}/var/log/ranger/usersync \
+        ${CRAFT_PART_INSTALL}/etc/ranger
+
+      # Unpack usersync tar
+      tar xvfz \
+        ${CRAFT_STAGE}${RANGER_TARGET}/ranger-${VERSION}-usersync.tar.gz \
+        --directory=${CRAFT_PART_INSTALL}${RANGER_HOME}/usersync \
+        --strip-components=1
+    stage:
+      - usr/lib/ranger/usersync
+      - var/log/ranger/usersync
+      - etc/ranger
+    permissions:
+      - path: usr/lib/ranger/usersync
+        owner: 584792
+        group: 584792
+        mode: "755"
+      - path: var/log/ranger/usersync
+        owner: 584792
+        group: 584792
+        mode: "755"
+      - path: etc/ranger
+        owner: 584792
+        group: 584792
+        mode: "755"
+
+  local-files:
+    plugin: dump
+    source: ./helper-scripts
+    organize:
+      admin-entrypoint.sh: home/ranger/scripts/ranger-admin-entrypoint.sh
+      usersync-entrypoint.sh: home/ranger/scripts/ranger-usersync-entrypoint.sh
+    stage:
+      - home/ranger/scripts/ranger-admin-entrypoint.sh
+      - home/ranger/scripts/ranger-usersync-entrypoint.sh
+    permissions:
+      - path: home/ranger/scripts
+        owner: 584792
+        group: 584792
+        mode: "755"
+
+  package-management:
+    plugin: nil
+    after: [admin, usersync]
+    overlay-packages:
+      - ca-certificates
+      - python3
+    overlay-script: |
+      ln -s /usr/bin/python3 ${CRAFT_OVERLAY}/usr/bin/python
```

Comment:
instead of doing this I think you can just include the `python-is-python3` overlay package